lib/modules/showImages.js

```diff
@@ -1032,11 +928,16 @@ function generateGallery(options) {
 		revealPiece(pieces[newIndex]);
 
 		if (module.options.conserveMemory.value) {
+			const first = newIndex - preloadCount;
+			const last = newIndex + preloadCount;
+
 			pieces.filter(piece => piece.media)
 				.map(piece => ({ media: piece.media, data: piece }))
 				::lazyUnload(piece => {
 					const index = pieces.indexOf(piece);
-					return index >= newIndex - preloadCount && index <= newIndex + preloadCount;
+					if (last > pieces.length && last % pieces.length >= index) return true;
+					if (first < 0 && ((first % pieces.length) + pieces.length) % pieces.length <= index) return true;
```

Comment:
`((first % pieces.length) + pieces.length) % pieces.length`

This might deserve its own function in utils/math.js. Maybe `positiveModulo`?